lib/manager/components/transform/AddCustomFile.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React, { Component } from 'react'
+
+import type { AddCustomFileProps, ReplaceFileFromStringFields, TransformProps } from '../../../types'
+
+import CustomCSVForm from './CustomCSVForm'
+
+/**
+ * Component that renders fields for AddCustomFile. This transformation shares csvData props with the ReplaceFileFromString transformation.
+ * TODO: adapt this transformation to include a file upload for larger custom files?
+ */
+export default class AddCustomFile extends Component<TransformProps<AddCustomFileProps>, AddCustomFileProps> {
+  constructor (props: TransformProps<AddCustomFileProps>) {
+    super(props)
+    this.state = {csvData: props.transformation.csvData, table: props.transformation.table}
+  }
+
+  // $FlowFixMe: Flow doesn't recognize return type for these arrow functions
+  _handleChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({...this.state, table: evt.target.value})
+  }
+
+  // $FlowFixMe[signature-verification-failure]
+  _onSaveCsvData = () => {
+    const csvData = this.state.csvData || null
+    const table = this.state.table || null
+    this.props.onSave({csvData, table}, this.props.index)
+  }
+
+  // $FlowFixMe[signature-verification-failure]
+  _onChangeCsvData = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const newState = {...this.state, csvData: evt.target.value}
+    this.setState(newState)
+    this._updateErrors(newState)
+  }
+
+  _getValidationErrors (fields: ReplaceFileFromStringFields): Array<string> {
+    const issues = []
+    const { csvData } = fields
+
+    // CSV data must be defined.
+    if (!csvData || csvData.length === 0) {
+      issues.push('CSV data must be defined.')
```

Comment:
This is bad practice. Can the issues array contain ids that get mapped to translation strings?

Reply:
Also easier to debug

Reply:
Overhauled the whole error handling in https://github.com/ibi-group/datatools-ui/pull/950/commits/31a86b78a00b56b725b96d0058361e5f7c64832b